delta/plugins/elasticsearch/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/elasticsearch/client/ElasticSearchClient.scala

```diff
@@ -0,0 +1,260 @@
+          )
+        }
+    }
+
+  /**
+    * Search for the provided ''query'' inside the ''indices''
+    *
+    * @param query        the initial search query
+    * @param indices      the indices to use on search (if empty, searches in all the indices)
+    * @param page         the pagination information
+    * @param fields       the fields to be returned
+    * @param sort         the sorting criteria
+    * @param qp           the optional query parameters
+    */
+  def search(
+      query: JsonObject,
+      indices: Set[String] = Set.empty,
+      qp: Query = Query(ignoreUnavailable -> "true", allowNoIndices -> "true")
+  )(
+      page: Pagination,
+      totalHits: Boolean = true,
+      fields: Set[String] = Set.empty,
+      sort: SortList = SortList.empty
+  ): HttpResult[SearchResults[JsonObject]] = {
+    val searchEndpoint = (endpoint / indexPath(indices) / searchPath).withQuery(qp)
+    val payload        = QueryBuilder(query).withPage(page).withFields(fields).withSort(sort).withTotalHits(totalHits).build
+    client.fromJsonTo[SearchResults[JsonObject]](Post(searchEndpoint, payload))
+  }
+
+  /**
+    * Search for the provided ''query'' inside the ''indices''
+    *
+    * @param query        the initial search query
+    * @param indices      the indices to use on search (if empty, searches in all the indices)
+    * @param page         the pagination information
+    * @param fields       the fields to be returned
+    * @param sort         the sorting criteria
+    * @param qp           the optional query parameters
+    */
+  def searchRaw(
+      query: JsonObject,
+      indices: Set[String] = Set.empty,
+      qp: Query = Query(ignoreUnavailable -> "true", allowNoIndices -> "true")
+  )(
+      page: Pagination,
+      totalHits: Boolean = true,
+      fields: Set[String] = Set.empty,
+      sort: SortList = SortList.empty
+  ): HttpResult[Json] = {
+    val searchEndpoint = (endpoint / indexPath(indices) / searchPath).withQuery(qp)
+    val payload        = QueryBuilder(query).withPage(page).withFields(fields).withSort(sort).withTotalHits(totalHits).build
+    client.toJson(Post(searchEndpoint, payload)).onErrorRecoverWith { err =>
+      err.jsonBody.map(IO.pure).getOrElse(IO.raiseError(err))
+    }
+  }
+
+  private def discardEntity(resp: HttpResponse) =
+    IO.delay(resp.discardEntityBytes()).hideErrors >> IO.unit
+
+  private def indexPath(indices: Set[String]): String =
+    if (indices.isEmpty) allIndexPath
+    else indices.map(sanitize(_, allowWildCard = true)).mkString(",")
+
+  /**
+    * Replaces the characters ' "\<>|,/?' in the provided index with '_' and drops all '_' prefixes.
+    * The wildcard (*) character will be only dropped when ''allowWildCard'' is set to false.
+    *
+    * @param index the index name to sanitize
+    * @param allowWildCard flag to allow wildcard (*) or not.
+    */
+  private def sanitize(index: String, allowWildCard: Boolean): String = {
```

Comment:
Instead of sanitizing here, we could maybe pass something other than a string when providing an index.
So that the caller has to behave himself a minimum when passing an index and no magic happens here